### PR TITLE
Customize cloud build region

### PR DESCRIPTION
**Pull request by DonggeLiu**
Avoids hardcoding cloud build region to `us-central1`, so that users (e.g., `target_experiment.py`) can define their own region.

### CODE REVIEW

**Comment by DonggeLiu** [comment]
This is why [the latest OFG exp](https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-10-14-661-dg-comparison/index.html) had 0 coverage: All failed because I was trying to use build pool not in `us-central1`.

File: infra/build/functions/build_lib.py [hunk 1]
```diff
@@ -96,8 +96,9 @@
     'GCB_BUILDPOOL_NAME', 'projects/oss-fuzz/locations/us-central1/'
     'workerPools/buildpool-experiments')
 
-US_CENTRAL_CLIENT_OPTIONS = google.api_core.client_options.ClientOptions(
-    api_endpoint='https://us-central1-cloudbuild.googleapis.com/')
+CLOUD_BUILD_LOCATION = os.getenv('CLOUD_BUILD_LOCATIONS', 'us-central1')
```
Comment (oliverchang) [inline infra/build/functions/build_lib.py:99]: Should we make the env var singular?

**Review by oliverchang — APPROVED** [review]

File: infra/build/functions/build_lib.py (hunk 1 above)
Comment (DonggeLiu) [reply infra/build/functions/build_lib.py]: Yep! Thanks!

**Comment by DonggeLiu** [comment]
I will merge this and launch another cloud exp in OFG

**Pull request merged**